tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -331,3 +331,57 @@ void test_mem_domain_migration(void)
 
 	k_thread_join(&child_thread, K_FOREVER);
 }
+
+K_MEM_PARTITION_DEFINE(overlap_part, ro_buf, sizeof(ro_buf),
+		       K_MEM_PARTITION_P_RW_U_RW);
+
+/**
+ * @brief Test system assert when new partition overlaps the existing partition
+ *
+ * @details
+ * Test Objective:
+ * - Test assertion if the new partition overlaps existing partition in domain
+ *
+ * Testing techniques:
+ * - System testing
+ *
+ * Prerequisite Conditions:
+ * - Before beginning that test only one partition ro_part must be placed
+ *   in the memory domain. As a result, in the current memory domain must left
+ *   empty space to add only one new memory partition.
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# Define testing memory partition overlap_part
+ * -# Change start address of the overlap_part by replacing default start
+ *  address with the start address of the ro_part partiiton. That way will be
+ *  possible to overlap partitions with each other in the memory domain.
+ * -# Try to add overlap_part to the memory domain. When adding the new
+ *  partition to the memory domain the system will assert that new partition
+ *  overlaps with the existing partition ro_part .
+ *
+ * Expected Test Result:
+ * - Must happen an assertion error indicating that the new partition overlaps
+ *   the existing one.
+ *
+ * Pass/Fail Criteria:
+ * - Success if the overlap assertion will happen.
+ * - Failure if the overlap assertion will not happen.
+ *
+ * Assumptions and Constraints:
+ * - N/A
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_mem_domain_add_partition()
+ */
+void test_mem_part_overlap(void)
+{
+	set_fault_valid(true);
+
+	overlap_part.start = ro_part.start;
```

Comment:
It seems like you already define the start address of overlap_part as the same as ro_part, is this necessary here?

Reply:
Yes, right, idea of the test is try to overlap two partitions with each other. If new partition will have same start address as existing one, when you will try to add it to the memory domain will happen an overlap, because one partitions will be trying to lay over existing parititon in the memory domain.

Reply:
I understood. I just thought at the moment you when defined K_MEM_PARTITION_DEFINE(overlap_part, ro_buf, sizeof(ro_buf), K_MEM_PARTITION_P_RW_U_RW),  the overlap_part.start has are been ro_part.start already? If you comment this line, will the oops still be happened?

Reply:
Oops won't happen, because new partition will go to the memory domain's free place. That memory domain can store 2 partitions.

Reply:
Now I understand your concerns. Because kernel oops will happen even if line `overlap_part.start = ro_part.start;` uncommented.